### ISSUE TITLE
[4.2.2] czmq `make check` tests do not display properly in terminal

### ISSUE THREAD

**Issue by BHare1985** [zeromq/czmq#2280]
Compiling czmq and running `make check` on Ubuntu 22.04 / Debian 11 / Debian 12 results in odd test output where the results are output before the tests are properly ran, see: https://gist.github.com/BHare1985/7469639aa05d7d580d9c8d602b08a0e5

Screenshot:
![WindowsTerminal_tI7OCu7Svo](https://github.com/zeromq/czmq/assets/2180284/b9ac9da6-8b05-40f9-be75-025896b47e47)

The issue does not happen when running tests with `libzmq`:

![WindowsTerminal_jJAN6oOTXE](https://github.com/zeromq/czmq/assets/2180284/1a7f1670-d941-460b-8d71-62c4593f029f)

**Comment by sphaero** [comment]
yes this is known. Just nobody has fixed it yet. Any idea where the problem origins?
